service/handlers/decks.js

```diff
@@ -0,0 +1,102 @@
+import { User } from '../models/User.js'
+import {validationResult} from 'express-validator'
+
+export const deckById = async (req, res) => {
+  const userId = req.headers.user
+  console.log(`user: ${userId} deckId ${req.params.id}`)
+  try {
+    const user = await User.findById(userId)
+    const deck = user.decks.id(req.params.id)
+    if (deck) {
+      res.send(deck)
+    } else {
+      res.sendStatus(404)
+    }
+  } catch (err) {
+    console.log(`${deckById.name}: ${err}`)
+    res.sendStatus(500)
+  }
+}
+
+export const getDecks = async (req, res) => {
+  const userId = req.headers.user
+  try {
+    const user = await User.findById(userId)
+    if (user) {
+      res.send(user.decks)
+    } else {
+      res.sendStatus(404)
+    }
+  } catch (err) {
+    console.log(`${getDecks.name}: ${err}`)
+    res.sendStatus(500)
+  }
+}
+
+export const createDeck = async (req, res) => {
+  const userId = req.headers.user
+  const newDeck = req.body
+  try {
+    const user = await User.findById(userId)
+    user.decks.push({
+      name: newDeck.name,
+      cards: []
+    })
+    await user.save()
+    res.sendStatus(204)
+  } catch (err) {
+    console.log(`${createDeck.name}: ${err}`)
+    res.sendStatus(500)
+  }
+}
+
+export const createCard = async (req, res) => {
+  const userId = req.headers.user
+  const deckId = req.params.id
+  const newCard = req.body
+  const errors = validationResult(req)
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array() })
```

Comment:
Why might it be a bad idea to return the contents of the errors directly to the user/

Reply:
We dont want the user to have access or knowledge to all information?